File: 프로그래머스/Java/2/12924. 숫자의 표현/숫자의 표현.java
class Solution {
    public int solution(int n) {
        int answer = 1;
        
        for(int i = 1; i <= n / 2; i++) {
            int sum = n;
            
            for(int j = i; j <= n; j++) {
                if(sum > 0) {
                    sum -= j;
                } else {
                    break;
                }
            }
            
            if(sum == 0) {
                answer++;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n) {
    int answer = 1;
    
    for(int start = 1; start <= n / 2; start++) {
        int sum = 0;
        for(int i = start; i <= n; i++) {
            sum += i;
            
            if(sum == n) {
                answer++;
                break;
            } else if (sum > n) {
                break;
            }
        }
    }
    return answer;
}